### DISCUSSION TITLE
What's up with pages that are not loading (when using nuxtlink)

### DISCUSSION THREAD

**Discussion by njiels** [nuxt/nuxt#21947] [Questions]
When using Nuxtlink pages are not being loaded directly. New URL's will be passed through, but you have to refresh manually to actually enforce them. This has caused me a lot of problems, and the weird thing is that it also seems to happen at random stages in a project's development. E.G. I've almost finished my portfolio website and everything was working fine, now suddenly i have to manually refresh when navigating to another page. When I change the nuxtlink elements to regular anchor elements it seems to be fixed, but still it bugs me to no end. Is there a clear and simple fix for this? It was already an issue in nuxt2 and I just don't get it 😅

**Comment by warflash** [comment]
Never seen that before. Can you check your console for errors that occur when clicking the nuxtlink? If there is none then a stackblitz or general reproduction would be great to investigate

**Comment by njiels** [comment]
```
[Vue warn]: Failed to resolve component: transition-slide
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
  at <ContentRenderer value=null > 
  at <[...Slug] onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > 
  at <RouteProvider key="/project/hiddetuinte" routeProps= {Component: {…}, route: {…}}Component: {__v_isVNode: true, __v_skip: true, type: {…}, props: {…}, key: null, …}route: {fullPath: '/project/hiddetuinte', path: '/project/hiddetuinte', query: {…}, hash: '', name: 'project-_slug', …}[[Prototype]]: Object pageKey="/project/hiddetuinte"  ... > 
  at <BaseTransition onAfterLeave= [ƒ] mode="out-in" appear=false  ... > 
  at <Transition onAfterLeave= [ƒ] name="page" mode="out-in" > 
  at <RouterView name=undefined route=undefined > 
  at <NuxtPage> 
  at <App key=3 > 
  at <NuxtRoot>
w
```

**Comment by njiels** [comment]
I solved the previous error (it had something to do with another issue on my website). Now I get this error:
`/Users/nielsvanleenen/Documents/Opdrachten/portfolio2023/portfolio_web/pages/index.vue` does not have a single root node and will cause errors when navigating between routes.

**Reply by warflash** [reply]
Well did you make sure your index file only has a single root element?

**Reply by njiels** [reply]
it seems to not be linked to the initial state of the code that caused the bug

**Comment by njiels** [comment]
Similar Error reappeared, more in line with what I first experienced. Mind you at other websites If worked on it could be that previous errors prevented me from collecting more helpful warnings.

Let me recreate the situation: I click on a very simple nuxt-link element (like below)
`<nuxt-link class="btn" to="/aboutme">About me</nuxt-link>`

It will render the page but with the following errors:

> chunk-3Q27KRJ5.js:97 [Vue warn]: Unhandled error during execution of render function 
>   at <ContentRenderer value=null > 
>   at <Index onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > 
>   at <RouteProvider key="/aboutme" routeProps= {Component: {…}, route: {…}} pageKey="/aboutme"  ... > 
>   at <BaseTransition onAfterLeave= [ƒ] mode="out-in" appear=false  ... > 
>   at <Transition onAfterLeave= [ƒ] name="page" mode="out-in" > 
>   at <RouterView name=undefined route=undefined > 
>   at <NuxtPage key="/aboutme" > 
>   at <App key=3 > 
>   at <NuxtRoot>
> warn @ chunk-3Q27KRJ5.js:97
> logError @ chunk-3Q27KRJ5.js:271
> handleError @ chunk-3Q27KRJ5.js:263
> renderComponentRoot @ chunk-3Q27KRJ5.js:909
> componentUpdateFn @ chunk-3Q27KRJ5.js:5747
> run @ reactivity.esm-bundler.js:178
> instance.update @ chunk-3Q27KRJ5.js:5860
> setupRenderEffect @ chunk-3Q27KRJ5.js:5868
> mountComponent @ chunk-3Q27KRJ5.js:5658
> processComponent @ chunk-3Q27KRJ5.js:5611
> patch @ chunk-3Q27KRJ5.js:5084
> mountChildren @ chunk-3Q27KRJ5.js:5328
> mountElement @ chunk-3Q27KRJ5.js:5235
> processElement @ chunk-3Q27KRJ5.js:5200
> patch @ chunk-3Q27KRJ5.js:5072
> componentUpdateFn @ chunk-3Q27KRJ5.js:5754
> run @ reactivity.esm-bundler.js:178
> instance.update @ chunk-3Q27KRJ5.js:5860
> setupRenderEffect @ chunk-3Q27KRJ5.js:5868
> mountComponent @ chunk-3Q27KRJ5.js:5658
> processComponent @ chunk-3Q27KRJ5.js:5611
> patch @ chunk-3Q27KRJ5.js:5084
> componentUpdateFn @ chunk-3Q27KRJ5.js:5754
> run @ reactivity.esm-bundler.js:178
> instance.update @ chunk-3Q27KRJ5.js:5860
> setupRenderEffect @ chunk-3Q27KRJ5.js:5868
> mountComponent @ chunk-3Q27KRJ5.js:5658
> processComponent @ chunk-3Q27KRJ5.js:5611
> patch @ chunk-3Q27KRJ5.js:5084
> mountSuspense @ chunk-3Q27KRJ5.js:1171
> process @ chunk-3Q27KRJ5.js:1117
> patch @ chunk-3Q27KRJ5.js:5109
> componentUpdateFn @ chunk-3Q27KRJ5.js:5754
> run @ reactivity.esm-bundler.js:178
> instance.update @ chunk-3Q27KRJ5.js:5860
> setupRenderEffect @ chunk-3Q27KRJ5.js:5868
> mountComponent @ chunk-3Q27KRJ5.js:5658
> processComponent @ chunk-3Q27KRJ5.js:5611
> patch @ chunk-3Q27KRJ5.js:5084
> componentUpdateFn @ chunk-3Q27KRJ5.js:5754
> run @ reactivity.esm-bundler.js:178
> instance.update @ chunk-3Q27KRJ5.js:5860
> setupRenderEffect @ chunk-3Q27KRJ5.js:5868
> mountComponent @ chunk-3Q27KRJ5.js:5658
> processComponent @ chunk-3Q27KRJ5.js:5611
> patch @ chunk-3Q27KRJ5.js:5084
> componentUpdateFn @ chunk-3Q27KRJ5.js:5754
> run @ reactivity.esm-bundler.js:178
> instance.update @ chunk-3Q27KRJ5.js:5860
> setupRenderEffect @ chunk-3Q27KRJ5.js:5868
> mountComponent @ chunk-3Q27KRJ5.js:5658
> processComponent @ chunk-3Q27KRJ5.js:5611
> patch @ chunk-3Q27KRJ5.js:5084
> componentUpdateFn @ chunk-3Q27KRJ5.js:5754
> run @ reactivity.esm-bundler.js:178
> instance.update @ chunk-3Q27KRJ5.js:5860
> setupRenderEffect @ chunk-3Q27KRJ5.js:5868
> mountComponent @ chunk-3Q27KRJ5.js:5658
> processComponent @ chunk-3Q27KRJ5.js:5611
> patch @ chunk-3Q27KRJ5.js:5084
> componentUpdateFn @ chunk-3Q27KRJ5.js:5819
> run @ reactivity.esm-bundler.js:178
> instance.update @ chunk-3Q27KRJ5.js:5860
> callWithErrorHandling @ chunk-3Q27KRJ5.js:213
> flushJobs @ chunk-3Q27KRJ5.js:411
> Promise.then (async)
> queueFlush @ chunk-3Q27KRJ5.js:324
> queuePostFlushCb @ chunk-3Q27KRJ5.js:344
> queueEffectWithSuspense @ chunk-3Q27KRJ5.js:1658
> scheduler @ chunk-3Q27KRJ5.js:1827
> triggerEffect @ reactivity.esm-bundler.js:373
> triggerEffects @ reactivity.esm-bundler.js:363
> triggerRefValue @ reactivity.esm-bundler.js:974
> (anonymous) @ reactivity.esm-bundler.js:1135
> triggerEffect @ reactivity.esm-bundler.js:373
> triggerEffects @ reactivity.esm-bundler.js:358
> triggerRefValue @ reactivity.esm-bundler.js:974
> (anonymous) @ reactivity.esm-bundler.js:1135
> triggerEffect @ reactivity.esm-bundler.js:373
> triggerEffects @ reactivity.esm-bundler.js:358
> triggerRefValue @ reactivity.esm-bundler.js:974
> set value @ reactivity.esm-bundler.js:1018
> finalizeNavigation @ vue-router.js:2570
> (anonymous) @ vue-router.js:2478
> Promise.then (async)
> pushWithRedirect @ vue-router.js:2446
> push @ vue-router.js:2372
> navigate @ vue-router.js:1628
> callWithErrorHandling @ chunk-3Q27KRJ5.js:213
> callWithAsyncErrorHandling @ chunk-3Q27KRJ5.js:221
> invoker @ chunk-MNWBZQKJ.js:322
> Show 20 more frames
> chunk-3Q27KRJ5.js:97 [Vue warn]: Unhandled error during execution of scheduler flush. This is likely a Vue internals bug. Please open an issue at https://new-issue.vuejs.org/?repo=vuejs/core 
>   at <ContentRenderer value=null > 
>   at <Index onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > 
>   at <RouteProvider key="/aboutme" routeProps= {Component: {…}, route: {…}} pageKey="/aboutme"  ... > 
>   at <BaseTransition onAfterLeave= [ƒ] mode="out-in" appear=false  ... > 
>   at <Transition onAfterLeave= [ƒ] name="page" mode="out-in" > 
>   at <RouterView name=undefined route=undefined > 
>   at <NuxtPage key="/aboutme" > 
>   at <App key=3 > 
>   at <NuxtRoot>
> warn @ chunk-3Q27KRJ5.js:97
> logError @ chunk-3Q27KRJ5.js:271
> handleError @ chunk-3Q27KRJ5.js:263
> callWithErrorHandling @ chunk-3Q27KRJ5.js:215
> flushJobs @ chunk-3Q27KRJ5.js:411
> Promise.then (async)
> queueFlush @ chunk-3Q27KRJ5.js:324
> queuePostFlushCb @ chunk-3Q27KRJ5.js:344
> queueEffectWithSuspense @ chunk-3Q27KRJ5.js:1658
> scheduler @ chunk-3Q27KRJ5.js:1827
> triggerEffect @ reactivity.esm-bundler.js:373
> triggerEffects @ reactivity.esm-bundler.js:363
> triggerRefValue @ reactivity.esm-bundler.js:974
> (anonymous) @ reactivity.esm-bundler.js:1135
> triggerEffect @ reactivity.esm-bundler.js:373
> triggerEffects @ reactivity.esm-bundler.js:358
> triggerRefValue @ reactivity.esm-bundler.js:974
> (anonymous) @ reactivity.esm-bundler.js:1135
> triggerEffect @ reactivity.esm-bundler.js:373
> triggerEffects @ reactivity.esm-bundler.js:358
> triggerRefValue @ reactivity.esm-bundler.js:974
> set value @ reactivity.esm-bundler.js:1018
> finalizeNavigation @ vue-router.js:2570
> (anonymous) @ vue-router.js:2478
> Promise.then (async)
> pushWithRedirect @ vue-router.js:2446
> push @ vue-router.js:2372
> navigate @ vue-router.js:1628
> callWithErrorHandling @ chunk-3Q27KRJ5.js:213
> callWithAsyncErrorHandling @ chunk-3Q27KRJ5.js:221
> invoker @ chunk-MNWBZQKJ.js:322
> Show 12 more frames
> index.vue:14 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'skills')
>     at index.vue:14:61
>     at Proxy.renderFnWithContext (chunk-3Q27KRJ5.js:823:13)
>     at Proxy.render (ContentRenderer.vue:57:27)
>     at renderComponentRoot (chunk-3Q27KRJ5.js:872:16)
>     at ReactiveEffect.componentUpdateFn [as fn] (chunk-3Q27KRJ5.js:5747:46)
>     at ReactiveEffect.run (reactivity.esm-bundler.js:178:19)
>     at instance.update (chunk-3Q27KRJ5.js:5860:52)
>     at setupRenderEffect (chunk-3Q27KRJ5.js:5868:5)
>     at mountComponent (chunk-3Q27KRJ5.js:5658:5)
>     at processComponent (chunk-3Q27KRJ5.js:5611:9)

Mind that the last error is displaying because It is a piece of code that is rendered in onMounted(). For some reason onMounted will load before DOM is correctly (re)loaded with this page. I've had it on previous pages and used this piece of code
`<NuxtPage :key="$route.fullPath" />` in app.vue. This works at all pages except for the aboutme page that I created after adjusting app.vue .

When I click on the back button `<a class="btn" href="" @click.prevent="$router.back()">` I will receive the following error:

> chunk-3Q27KRJ5.js:97 [Vue warn]: Unhandled error during execution of scheduler flush. This is likely a Vue internals bug. Please open an issue at https://new-issue.vuejs.org/?repo=vuejs/core 
>   at <App key=3 > 
>   at <NuxtRoot>
> warn @ chunk-3Q27KRJ5.js:97
> logError @ chunk-3Q27KRJ5.js:271
> handleError @ chunk-3Q27KRJ5.js:263
> callWithErrorHandling @ chunk-3Q27KRJ5.js:215
> flushJobs @ chunk-3Q27KRJ5.js:411
> Promise.then (async)
> queueFlush @ chunk-3Q27KRJ5.js:324
> queueJob @ chunk-3Q27KRJ5.js:318
> (anonymous) @ chunk-3Q27KRJ5.js:5856
> triggerEffect @ reactivity.esm-bundler.js:373
> triggerEffects @ reactivity.esm-bundler.js:363
> triggerRefValue @ reactivity.esm-bundler.js:974
> (anonymous) @ reactivity.esm-bundler.js:1135
> triggerEffect @ reactivity.esm-bundler.js:373
> triggerEffects @ reactivity.esm-bundler.js:358
> triggerRefValue @ reactivity.esm-bundler.js:974
> (anonymous) @ reactivity.esm-bundler.js:1135
> triggerEffect @ reactivity.esm-bundler.js:373
> triggerEffects @ reactivity.esm-bundler.js:358
> triggerRefValue @ reactivity.esm-bundler.js:974
> (anonymous) @ reactivity.esm-bundler.js:1135
> triggerEffect @ reactivity.esm-bundler.js:373
> triggerEffects @ reactivity.esm-bundler.js:358
> triggerRefValue @ reactivity.esm-bundler.js:974
> (anonymous) @ reactivity.esm-bundler.js:1135
> triggerEffect @ reactivity.esm-bundler.js:373
> triggerEffects @ reactivity.esm-bundler.js:358
> triggerRefValue @ reactivity.esm-bundler.js:974
> set value @ reactivity.esm-bundler.js:1018
> finalizeNavigation @ vue-router.js:2570
> (anonymous) @ vue-router.js:2625
> Promise.then (async)
> (anonymous) @ vue-router.js:2624
> (anonymous) @ vue-router.js:417
> popStateHandler @ vue-router.js:416
> Show 20 more frames
> chunk-MNWBZQKJ.js:82 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'parentNode')
>     at parentNode (chunk-MNWBZQKJ.js:82:30)
>     at ReactiveEffect.componentUpdateFn [as fn] (chunk-3Q27KRJ5.js:5823:11)
>     at ReactiveEffect.run (reactivity.esm-bundler.js:178:19)
>     at instance.update (chunk-3Q27KRJ5.js:5860:52)
>     at callWithErrorHandling (chunk-3Q27KRJ5.js:213:32)
>     at flushJobs (chunk-3Q27KRJ5.js:411:9)

**Reply by njiels** [reply]
This error is received when I navigate to aboutme directly from homepage. When i navigate to this page via a different trajectory (home page > project page > aboutme) It will work fine, except for the onMounted issue. It will give a different error and also will not work (in the example above it will give an error and also work fine)

**Reply by njiels** [reply]
After using this route, I went back to my homepage and tried to recreate the error. Everything suddenly worked fine! It only appears when I opened the website first in dev mode (have not checked build yet), or after a reload of the homepage. 🤔
